server_manager/www/shadowbox_server.ts

```diff
@@ -149,22 +150,64 @@ export class ShadowboxServer implements server.Server {
     await this.api.request<void>(`access-keys/${keyId}/data-limit`, 'DELETE');
   }
 
-  async getDataUsage(): Promise<server.BytesByAccessKey> {
+  async getServerMetrics(): Promise<server.ServerMetricsJson> {
+    if (
+      (await this.getSupportedExperimentalEndpoints()).has('server/metrics')
+    ) {
+      return this.api.request<server.ServerMetricsJson>(
+        'experimental/server/metrics?since=30d'
+      );
+    }
+
+    const result: server.ServerMetricsJson = {
+      server: [],
+      accessKeys: [],
+    };
+
     const jsonResponse =
       await this.api.request<DataUsageByAccessKeyJson>('metrics/transfer');
-    const usageMap = new Map<server.AccessKeyId, number>();
+
     for (const [accessKeyId, bytes] of Object.entries(
       jsonResponse.bytesTransferredByUserId
     )) {
-      usageMap.set(accessKeyId, bytes ?? 0);
+      result.accessKeys.push({
+        accessKeyId: Number(accessKeyId),
+        dataTransferred: {bytes},
+      });
     }
-    return usageMap;
+
+    return result;
   }
 
   getName(): string {
     return this.serverConfig?.name;
   }
 
+  async getSupportedExperimentalEndpoints(): Promise<Set<string>> {
+    if (this._supportedExperimentalEndpointCache)
+      return this._supportedExperimentalEndpointCache;
```

Comment:
nit: [Always use braces for control flow blocks.](https://google.github.io/styleguide/tsguide.html#control-flow-statements-blocks)

```suggestion
    if (this._supportedExperimentalEndpointCache) {
      return this._supportedExperimentalEndpointCache;
    }
```